lib/operations/distinct.js

```diff
@@ -73,14 +66,24 @@ class DistinctOperation extends OperationBaseV2 {
       return callback(err, null);
     }
 
-    // Execute the command
-    executeCommand(coll.s.db, cmd, options, (err, result) => {
+    const ns =
+      typeof coll.s.db.namespace === 'string'
+        ? `${coll.s.db.namespace}.$cmd`
+        : coll.s.db.namespace.withCollection('$cmd');
+
+    server.command(ns, cmd, options, (err, result) => {
       if (err) return handleCallback(callback, err);
+
+      result = result.result;
```

Comment:
What if we have `options.full`? `executeCommand` would do `if (options.full) return handleCallback(callback, null, result);`

Reply:
I've switched this to use `CommandOperationV2` now, which supports `full` again. There is a glaring bug in it though, in that the `DistinctOperation` expects a `result.values` to exist - but this will not exist if a full response was requested. Open to suggestions here.

Reply:
Is this a new bug or did it exist before?

Reply:
Fixed this by checking for `full` in the `DistinctOperation.execute` method. There might be a more streamlined way to avoid this in the future, but I think this is fine for now given that `distinct` is doing something different with the return value.